### PR TITLE
Remove `Polaris.VERSION` from the global window object

### PR DESCRIPTION
**Pull request by aaronccasanova**
### WHAT is this pull request doing?

This PR removes the `Polaris.VERSION` variable injected on the global window object. From my understanding this has always been used for internal applications and is already queryable via the `--polaris-version-number` custom property. Additionally, this clean up allows us to remove `configure.ts` from the `package.json` `sideEffects` and build validation script.

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris/blob/main/README.md#local-development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris/blob/main/.github/CONTRIBUTING.md#changelog)

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import React from 'react';
import {Page} from '../src';

export function Playground() {
  return (
    <Page title="Playground">
      {/* Add the code you want to test in here */}
    </Page>
  );
}
```

</details>

### 🎩 checklist

- [ ] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

### CODE REVIEW

**Review by alex-page — APPROVED** [review]
Should this point at v11?

File: .changeset/light-steaks-promise.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@shopify/polaris': minor
```
Comment (aaronccasanova) [inline .changeset/light-steaks-promise.md]:
```suggestion
'@shopify/polaris': major
```

**Comment by sophschneider** [comment]
can we remove the [`replace` from rollup](https://github.com/Shopify/polaris/blob/main/polaris-react/rollup.config.mjs#L39-L43) now too?

**Comment by aaronccasanova** [comment]
> can we remove the [replace from rollup](https://github.com/Shopify/polaris/blob/main/polaris-react/rollup.config.mjs#L39-L43) now too?

@sophschneider The replacement is still needed for the [--polaris-version-number](https://github.com/Shopify/polaris/blob/efdcb10dbf1ac7a29b33f5e5e008ca1e011e5275/polaris-react/src/components/AppProvider/AppProvider.scss#L5) custom property

**Review by BPScott — APPROVED** [review]
My main concern was removing the global value on `window` and I was ok leaving the `polarisVersion` named export.

If others don't see the value in the named export and want to remove that then I'm happy too

**Review by sam-b-rose — APPROVED** [review]

**Review by sophschneider — APPROVED** [review]

**Review by tmlayton — APPROVED** [review]

**Comment by aaronccasanova** [comment]
> My main concern was removing the global value on window and I was ok leaving the polarisVersion named export.
> If others don't see the value in the named export and want to remove that then I'm happy too

@BPScott Yeah, it felt extraneous to me as it's already accessible from the package.json:

image

**Pull request merged**